js/script.js

```diff
@@ -53,74 +53,73 @@ profileApp.validator = {
-            this.officePin.value = this.homePin.value;
-            this.officePhone.value = this.homePhone.value;
-            this.officeFax.value = this.homeFax.value;
+        if (this.checkboxAddress.prop('checked')) {
+            $('#officeAddress').prop('disabled', true);
+            this.officeStreet.val(this.homeStreet.val());
+            this.officeCity.val(this.homeCity.val());
+            this.officeState.val(this.homeState.val());
+            this.officePin.val(this.homePin.val());
+            this.officePhone.val(this.homePhone.val());
+            this.officeFax.val(this.homeFax.val());
         }
         else {
-            document.getElementById('officeAddress').disabled = false;
+            $('#officeAddress').prop('disabled', false);
         }
     },
     validateInputElements: function () {
-        var inputElementLength = this.inputElements.length;
-        for (var i = 0; i < inputElementLength; i++) {
-            var currentParentElement = this.inputElements[i].parentElement;
+        var appObj = this;
+        this.inputElements.each(function () {
+            var currentParentElement = $(this).parent();
 
-            if (!this.validateInput(this.inputElements[i]) && (!currentParentElement.classList.contains('has-error'))) {
-                this.addErrorClass(currentParentElement);
+            if (!appObj.validateInput(this) && !currentParentElement.hasClass('has-error')) {
+                appObj.addErrorClass(currentParentElement);
             }
-            else if (this.validateInput(this.inputElements[i]) && (currentParentElement.classList.contains('has-error'))) {
-                this.removeErrorClass(currentParentElement);
+            else if (appObj.validateInput(this) && currentParentElement.hasClass('has-error')) {
+                appObj.removeErrorClass(currentParentElement);
             }
+        });
 
+        if (!appObj.flag && $("span").filter("#helpBlock").length) {
+            var helpBlock = $('#helpBlock');
+            helpBlock.parent().filter('#helpBlock').remove();
         }
-
-        if (!this.flag) {
-            var helpBlock = document.getElementById('helpBlock');
-            
-            if (helpBlock) {
-                helpBlock.parentElement.removeChild(helpBlock);
-            }
-        }
-        else if (!document.getElementById('helpBlock')) {
-            var helpSpan = document.createElement("span");
-            helpSpan.innerHTML = '<span id="helpBlock" class="help-block text-danger block-center">Please correct the columns with errors.</span>';
-            var validate = document.getElementById('validate');
-            validate.parentElement.insertBefore(helpSpan, validate);
+        else if (appObj.flag && $("span").filter("#helpBlock").length === 0) {
+            var helpSpan = $('<span id="helpBlock" class="help-block text-danger block-center">Please correct the columns with errors.</span>');
+            var validate = $('#validate');
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove the declaration of the unused 'validate' variable. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://localhost:9000/coding_rules#rule_key=javascript%3AUnusedVariable)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this useless assignment to local variable "validate" [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://localhost:9000/coding_rules#rule_key=javascript%3AS1854)